cms/server/admin/templates/training_program_tasks.html

```
@@ -1,38 +1,40 @@
     {% include "fragments/overload_warning.html" %}
 
-    <!-- Add Task Card -->
-    <div class="tp-filter-card">
-        <label class="tp-filter-label" for="task_id">Add Task:</label>
-        <form action="{{ url("training_program", training_program.id, "tasks", "add") }}" method="POST" style="display: flex; gap: 12px; flex: 1;">
+    <!-- Toolbar: Search + Add -->
+    {{ task_search_toolbar(training_program.managing_contest.tasks|length, admin.permission_all and unassigned_tasks|length > 0) }}
+
+    {% if admin.permission_all and unassigned_tasks|length > 0 %}
+    <div class="tp-filter-card mb-4" id="add-task-panel" {% if training_program.managing_contest.tasks|length > 0 %}style="display:none;"{% endif %}>
+        <form class="is-flex is-align-items-center is-flex-wrap-wrap" style="gap: 12px; flex: 1;" action="{{ url("training_program", training_program.id, "tasks", "add") }}" method="POST">
```

HTMLHint (1.9.2)

[error] Special characters must be escaped : [ < ].

(spec-char-escape)